### PR TITLE
Fixes choiced moves not locked in after ability block

### LINKED ISSUE #5151: Choice Item Lock not considering Dazzling / Soundproof / Bulletproof
### Description

For example considering Dazzling, a Pokemon that locks itself into a Priority Move against a Pokemon with Dazzling (or Queenly Majesty) does not get locked into said Priority Move and can switch moves.

Clip: (Also tested with Soundproof, so I am assuming Bulletproof is gonna be the same)
https://github.com/user-attachments/assets/772521c3-236b-4b03-b681-3ca8fb643015

### Version

1.9.0 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

._.terra._.

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes  #5151

`HITMARKER_OBEYS` was set to late for the move to be locked due to a choice item. 

I removed the global because it isn't needed anymore since the obedience check will happen before the z move check. 

Test provided by @Pawkkie

### CODE REVIEW

File: include/battle.h [hunk 1]
```diff
@@ -802,7 +802,7 @@ struct BattleStruct
     u8 categoryOverride; // for Z-Moves and Max Moves
     u32 stellarBoostFlags[NUM_BATTLE_SIDES]; // stored as a bitfield of flags for all types for each side
     u8 fickleBeamBoosted:1;
-    u8 obedienceResult:3;
+    u8 padding:6;
```
Comment (Bassoonian) [inline include/battle.h:805]: Any reason we don't just move the padding to the end? It'd make it more clear where a new u8 starts

Comment (AlexOn1ine) [reply include/battle.h:805]: I would rather do that in upcoming to avoid conflicts.

Comment (AlexOn1ine) [reply include/battle.h]: actually I did it right now.

**Comment by hedara90** [comment]
I've looked over it, and the obedience part is fine.
I have a question about the obedience + choice items, when `DISOBEYS_RANDOM_MOVE` is the case, is the mon supposed to be locked into the randomly chosen move?

**Comment by AlexOn1ine** [comment]
> I've looked over it, and the obedience part is fine. I have a question about the obedience + choice items, when `DISOBEYS_RANDOM_MOVE` is the case, is the mon supposed to be locked into the randomly chosen move?

what should happen in vanilla?

**Comment by hedara90** [comment]
> what should happen in vanilla?

I checked on base `pokeemerald` and the same thing happened, so I'm judging it as correct.

**Review by hedara90 — APPROVED** [review]

**Pull request merged**